src/cryptonote_core/tx_pool.cpp

```diff
@@ -975,7 +977,7 @@ namespace cryptonote
         remaining_added_txids = std::vector<crypto::hash>(txids.begin() + max_tx_count, txids.end());
         txids.erase(txids.begin() + max_tx_count, txids.end());
       }
-      get_transactions_info(txids, added_txs, include_sensitive);
+      get_transactions_info(txids, added_txs, include_sensitive, limit_size);
```

Comment:
The limit we want to pass here should actually be `limit_size - 32ish * txids.size()` because we want to only use up the space for full-bodied txs that won't be used up by the TXIDs. 32*ish* because of encoding details on top of the 32 raw bytes per hash. If the cumulative tx blob size *would* be *much* greater than the limit then the cumulative size of the remaining TXIDs will push the total data size (full tx + txids) over the limit size.